src/Processors/Merges/Algorithms/RowRef.h

```diff
@@ -159,6 +159,26 @@ struct RowRef
         return true;
     }
 
+    static size_t checkEqualsFirstNonEqual(size_t size, size_t offset, const IColumn ** lhs, size_t lhs_row, const IColumn ** rhs, size_t rhs_row)
+    {
+        for (size_t i = 0; i < size; ++i)
+        {
+            auto col_number = (offset + i) % size;
```

Comment:
I'm afraid of this division - change to something like if (offset + i == size) offset = 0

Reply:
Not sure that simple reset to 0 will work

But something like, could.
But not sure that it's better

```
if (offset + i) == size)
   offset -= size

auto col_number = (offset + i)
```